feature/bgp/addpath/ate_tests/addpath/README.md

```diff
@@ -0,0 +1,91 @@
+* Ensure that the EBGP and IBGP peering are enabled for both address famailies.
+* DUT should be configured with "Both" Send and Receive ability for addpath on all IBGP peering. Same for ATE3:Port3, ATE4:Port4 and ATE5:Port5
+  * During BGP capabilities negotiation in Open message, verify that the DUT negotiated addpath cability with Send/Receive field set to "3"
+
+* Configure ATE1:Port1 and ATE2:Port2 to advertise same prefix "prefix-1" with different Protocol next-hops to DUT:Port1 and DUT:Port2
+  * Verify that the DUT is advertising multiple paths to prefix-1 to RRCs ATE3 and ATE4 as well as to the RRS ATE5 with different path-ids
+* Configure each of ATE1:Port1 and ATE2:Port2 to advertise prefix "prefix-1" with 2 different Protocol next-hops to DUT:Port1 and DUT:Port2 respectively. Then configure DUT to send maximum 3 paths over ADD-Path to ATE3 and ATE4
+  * Verify that ATE3 and ATE4 receive only 3 different paths (out of 4) w/ unique path-ids from DUT for prefix-1.
+* Configure RRCs ATE3:Port3 and ATE4:Port4 to advertise "prefix-2" to DUT with different Protocol next-hops.
+  * Verify that the DUT advertises multiple paths for prefix-2 to ATE5 with different path-ids
+* Configure RRCs ATE3:Port3 and ATE4:Port4 each to advertise 2 different paths for "prefix-2" to DUT with different Protocol next-hops each. Then configure DUT to send maximum 3 paths over ADD-Path to ATE5
+  * Verify that ATE5 receives only 3 different paths (out of 4) w/ unique path-ids from DUT for prefix-2.
+
+**Test-2**: Verify ADDPATH scaling 
+* DUT:Port3 and DUT:Port4 has IBGP peering with directly connected ATE3:Port3 and ATE4:Port4 respectively. In this case, DUT is the RR server and ATE3 and ATE4 are RR clients
+* DUT:Port5 has IBGP peering with directly connected ATE5:Port5. In this case DUT is the RR client and ATE5:Port5 is the RR server
+* Ensure that the EBGP and IBGP peering are enabled for both address famailies.
+* DUT is configued with addpath Receive capability only on its peering with ATE3:Port3 and ATE4:Port4
+  * During BGP capabilities negotiation in Open message, verify that the DUT negotiated addpath cability with Send/Receive field set to "1" signaling Receive capability.
+ * DUT is configued with addpath Send capability only on its peering with ATE5:Port5
+  * During BGP capabilities negotiation in Open message, verify that the DUT negotiated addpath cability with Send/Receive field set to "2" signaling send capability.
+ * Enable BGP multipath as well in the DUT.
+ * ATE3, 4 and 5 should be conffigured for addpath "both" capability.
+ * Configue ATE3 and ATE4 each to advertise 1M v4 prefixes with 32 different NHs for each prefix with a different path-id. Requirement here is for the DUT to receive 1M routes with 64 different NHs for the same prefixes but different path-id. Follow the same process for IPv6 with 600k routes and 32 distict NHs for each prefix from each ATE.
```

Comment:
OK fo 64M path in total.

BTW, the path-id mey ne not unique if it comes form different peer. I would expect that ATE3 and ATE4, both will actually sent path-d = 1.
This is OK, as it comse form disticnt sessions.

Reply:
That part is understood. But anyways I have tried to clarify